test/e2e/suites/apivalidations/openstackmachine_test.go

```diff
@@ -354,6 +354,152 @@ var _ = Describe("OpenStackMachine API validations", func() {
+		})
+
+		It("should not allow item with empty name", func() {
+			machine := defaultMachine()
+			machine.Spec.SchedulerHintAdditionalProperties = []infrav1.SchedulerHintAdditionalProperty{
+				{
+					Name: "",
+					Value: infrav1.SchedulerHintAdditionalValue{
+						Type: infrav1.SchedulerHintTypeBool,
+						Bool: ptr.To(false),
+					},
+				},
+			}
+			Expect(k8sClient.Create(ctx, machine)).NotTo(Succeed(), "Creating a machine with SchedulerHintAdditionalProperties including an item with empty name should fail.")
+		})
+
+		It("should not allow item with empty value", func() {
+			machine := defaultMachine()
+			machine.Spec.SchedulerHintAdditionalProperties = []infrav1.SchedulerHintAdditionalProperty{
+				{
+					Name: "test-hints",
+				},
+			}
+			Expect(k8sClient.Create(ctx, machine)).NotTo(Succeed(), "Creating a machine with SchedulerHintAdditionalProperties including an item with empty value should fail.")
+		})
+
+		It("should allow correct SchedulerHintAdditionalProperties", func() {
+			machineB := defaultMachine()
+			machineB.Spec.SchedulerHintAdditionalProperties = []infrav1.SchedulerHintAdditionalProperty{
+				{
+					Name: "test-hints",
+					Value: infrav1.SchedulerHintAdditionalValue{
+						Type: infrav1.SchedulerHintTypeBool,
+						Bool: ptr.To(true),
+					},
+				},
+			}
+			By("Creating SchedulerHint with bool type")
+			Expect(k8sClient.Create(ctx, machineB)).To(Succeed(), "Creating a machine with bool type scheduler hint property should succeed.")
+			machineN := defaultMachine()
+			machineN.Spec.SchedulerHintAdditionalProperties = []infrav1.SchedulerHintAdditionalProperty{
+				{
+					Name: "test-hints",
+					Value: infrav1.SchedulerHintAdditionalValue{
+						Type:   infrav1.SchedulerHintTypeNumber,
+						Number: ptr.To(1),
+					},
+				},
+			}
+			By("Creating SchedulerHint with number type")
+			Expect(k8sClient.Create(ctx, machineN)).To(Succeed(), "Creating a machine with number type scheduler hint property should succeed.")
+			machineS := defaultMachine()
+			machineS.Spec.SchedulerHintAdditionalProperties = []infrav1.SchedulerHintAdditionalProperty{
+				{
+					Name: "test-hints",
+					Value: infrav1.SchedulerHintAdditionalValue{
+						Type:   infrav1.SchedulerHintTypeString,
+						String: ptr.To("test-hint"),
+					},
+				},
+			}
+			By("Creating SchedulerHint with string type")
+			Expect(k8sClient.Create(ctx, machineS)).To(Succeed(), "Creating a machine with string type scheduler hint property should succeed.")
+		})
+
+		It("should not allow incorrect SchedulerHintAdditionalProperties", func() {
+			machineBN := defaultMachine()
+			machineBN.Spec.SchedulerHintAdditionalProperties = []infrav1.SchedulerHintAdditionalProperty{
+				{
+					Name: "test-hints",
+					Value: infrav1.SchedulerHintAdditionalValue{
+						Type:   infrav1.SchedulerHintTypeBool,
+						Number: ptr.To(1),
+					},
+				},
+			}
+			By("Creating SchedulerHint with bool type and number value")
```

Comment:
nit: could you make these separate tests with their own It()? This both allows them to run in parallel, and also enables ginkgo to separate out the logs if one of them fails.